src/toil/job.py

```diff
@@ -1077,6 +1116,16 @@ def writeGlobalFile(self, localFileName, cleanup=False):
                     fileHandle = open(absLocalFileName, 'r')
                     with self._pendingFileWritesLock:
                         self._pendingFileWrites.add(jobStoreFileID)
+                    # Before we can start the async process, we should also create a dummy harbinger
+                    # file in the cache such that any subsequent jobs asking for this file will not
+                    # attempt to download it from the job store till the write is complete.  We do
+                    # this now and then overwrite the harbinger with the true pid of the writing
+                    # thread because there is an edge case where readGlobalFile in a subsequent job
```

Comment:
This verbiage is out of date/misleading now that we established that all threads share the main threads PID.